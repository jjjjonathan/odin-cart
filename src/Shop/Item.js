import React from "react";

export default function Item(props) {
  return (
    <div className="item-card">
      <h4 className="item-title">{props.title}</h4>
      <p className="item-description">{props.description}</p>
      <button
        data-index={props.index}
        type="button"
        onClick={props.onAddToCartClick}
      >
        Add to cart
      </button>
    </div>
  );
}
