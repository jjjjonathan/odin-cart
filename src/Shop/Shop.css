h3 {
  text-align: center;
  font-size: 4rem;
}

h2,
h3,
.lead {
  margin-left: 20px;
  margin-right: 20px;
}
