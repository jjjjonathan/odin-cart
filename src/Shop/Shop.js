import React, { useState } from "react";
import Item from "./Item";

export default function Shop(props) {
  const [items, setItems] = useState([
    {
      title: "Odinfeathers",
      description: "Faux feathers in pastel gradients",
      price: 595,
    },
    {
      title: "The Nondescript Odinstone",
      description: "You expected description? See above",
      price: 9995,
    },
    {
      title: "Ode in Car",
      description: "The hit single. Vinyl single only",
      price: 39,
    },
    {
      title: "Odincart",
      description:
        "Virtual shopping cart. Contains no real products nor is it itself a real product",
      price: 299,
    },
  ]);

  return (
    <div>
      <div>
        <h3>Look for things to fill your Odincart.</h3>
        <p className="lead">Remember, none of this is real.</p>
      </div>
      <div className="shop">
        {items.map((item) => {
          return <Item title={item.title} description={item.description} />;
        })}
      </div>
    </div>
  );
}
