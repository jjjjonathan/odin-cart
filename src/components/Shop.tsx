import Item from './Item';

type ShopItem = {
  title: string;
  description: string;
  price: number;
  quantityInCart: number;
};

type ShopProps = {
  items: ShopItem[];
  onAddToCart: () => void;
  onRemoveFromCart: () => number;
};

const Shop = ({ items, onAddToCart, onRemoveFromCart }: ShopProps) => (
  <>
    <div>
      <h3>Look for things to fill your Odincart.</h3>
      <p className="lead">Remember, none of this is real.</p>
    </div>
    <div className="shop">
      {items.map((item, index) => {
        return (
          <Item
            title={item.title}
            description={item.description}
            price={item.price}
            onAddToCart={onAddToCart}
            onRemoveFromCart={onRemoveFromCart}
            index={index}
            key={index}
            quantityInCart={item.quantityInCart}
          />
        );
      })}
    </div>
  </>
);

export default Shop;
