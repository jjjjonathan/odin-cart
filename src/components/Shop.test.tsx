import Shop from './Shop';
import renderer from 'react-test-renderer';

describe('Shop Component', () => {
  it('should match snapshot', () => {
    const items = [
      {
        title: 'Odinfeathers',
        description: 'Faux feathers in pastel gradients',
        price: 595,
        quantityInCart: 0,
      },
      {
        title: 'The Nondescript Odinstone',
        description: 'If you expected description, See above',
        price: 9995,
        quantityInCart: 0,
      },
    ];

    const MockAddToCart = jest.fn();
    const MockRevomeFromCart = jest.fn();

    const tree = renderer
      .create(
        <Shop
          items={items}
          onAddToCart={MockAddToCart}
          onRemoveFromCart={MockRevomeFromCart}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
