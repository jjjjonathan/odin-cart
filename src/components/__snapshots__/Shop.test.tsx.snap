// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop Component should match snapshot 1`] = `
Array [
  <div>
    <h3>
      Look for things to fill your Odincart.
    </h3>
    <p
      className="lead"
    >
      Remember, none of this is real.
    </p>
  </div>,
  <div
    className="shop"
  >
    <div
      className="item-card"
    >
      <h4
        className="item-title"
      >
        Odinfeathers
      </h4>
      <p
        className="item-description"
      >
        Faux feathers in pastel gradients
      </p>
      <p
        className="price"
      >
        $
        595
      </p>
      <div
        className="item-form"
      >
        <form
          data-index={0}
          noValidate={true}
          onSubmit={[Function]}
        >
          <label
            htmlFor="quantity"
          >
            Qty:
          </label>
          <select
            id="quantity"
            name="quantity"
          >
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
          </select>
          <button
            type="submit"
          >
            Add to cart
          </button>
        </form>
      </div>
    </div>
    <div
      className="item-card"
    >
      <h4
        className="item-title"
      >
        The Nondescript Odinstone
      </h4>
      <p
        className="item-description"
      >
        If you expected description, See above
      </p>
      <p
        className="price"
      >
        $
        9995
      </p>
      <div
        className="item-form"
      >
        <form
          data-index={1}
          noValidate={true}
          onSubmit={[Function]}
        >
          <label
            htmlFor="quantity"
          >
            Qty:
          </label>
          <select
            id="quantity"
            name="quantity"
          >
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
          </select>
          <button
            type="submit"
          >
            Add to cart
          </button>
        </form>
      </div>
    </div>
  </div>,
]
`;
